### PR TITLE
Use the logging library for logs

### PR DESCRIPTION
**Pull request by sburba**

### CODE REVIEW

File: api/src/game_state_server.py [hunk 1]
```diff
@@ -103,7 +107,9 @@ def new_connection_request(self, client_id: Hashable, room_id: str) -> Message:
                     except (WrongTypeError, MissingValueError, TypeError) as e:
                         # Don't raise here. Loading a room minus any tokens that have
                         # been corrupted is still valuable.
-                        print(e)
+                        logger.exception(
```
Comment (sjohnson31) [inline api/src/game_state_server.py:110]: Here you use logger.exception, but on line 176 you use logger.warning. and it feels like it should be the same level.

File: api/src/game_state_server.py [hunk 2]
```diff
@@ -175,20 +182,25 @@ async def process_updates(
...
                     if not self._is_valid_token(token):
+                        logger.warning(f'Invalid token data {token}')
                         yield Message(
                             {client_id},
                             MessageContents(
                                 'error', f'Received a bad token: {data}', request_id
                             ),
                         )
-                    if not self._is_valid_position(token, room_id):
+                    if self._is_valid_position(token, room_id):
```
Comment (sjohnson31) [inline api/src/game_state_server.py:192]: Good bug fix, should be different PR. Side note: same bug exists just above this.

File: api/src/wsmanager.py [hunk 3]
```diff
@@ -86,7 +89,7 @@ async def consume(
         try:
             message = json.loads(json_message)
         except json.JSONDecodeError as e:
-            print(e)
+            logger.info("invalid json received from client", exc_info=e)
```
Comment (sjohnson31) [inline api/src/wsmanager.py:92]: These should be single quotes to be consistent.

File: api/src/game_state_server.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from dataclasses import dataclass
+import logging
```
Comment (sjohnson31) [inline api/src/game_state_server.py:2]: You use logging.getLogger(__name__) in wsmanager, but not here.

File: api/src/game_state_server.py [hunk 5]
```diff
@@ -30,9 +31,9 @@ def assign_colors(tokens: List[Token]) -> None:
     for token in tokens:
         if not token.color_rgb:
             if not available_colors:
-                print(f'Max colors reached for icon ' f'{content_id(token.contents)}')
+                logger.info(f'Max colors reached for icon {content_id(token.contents)}')
```
Comment (sjohnson31) [inline api/src/game_state_server.py:34]: logging.getLogger(__name__) should be above this.

File: api/src/game_state_server.py (hunk 1 above)
Comment (sburba) [reply api/src/game_state_server.py:110]: I think this is appropriate. On line 176 we got a bad message from a client, there's nothing wrong with the server. Here there's something wrong with the server because rooms are corrupted

File: api/src/wsmanager.py (hunk 3 above)
Comment (sburba) [reply api/src/wsmanager.py:92]: good catch, we should have a formatter check this ;)

File: api/src/game_state_server.py (hunk 2 above)
Comment (sburba) [reply api/src/game_state_server.py:192]: reverted for this pr

File: api/src/game_state_server.py (hunk 5 above)
Comment (sburba) [reply api/src/game_state_server.py:34]: good point, done

File: api/src/game_state_server.py (hunk 1 above)
Comment (sburba) [reply api/src/game_state_server.py:110]: On second thought though I'm going to change line 176 to logger.info because it's not really a warning if a client is sending invalid data, since anyone can do that any time
